### ISSUE TITLE
[0.3.8.5] NPC Age mismatch bug still present

### ISSUE THREAD

**Issue by serveral1** [Innoxia/liliths-throne-public#1361]
Fresh start. Using Lilaya as an example.
![image](https://user-images.githubusercontent.com/10162026/85934631-9e236d00-b8b3-11ea-8c6d-64f78e21efa7.png)
![image](https://user-images.githubusercontent.com/10162026/85934636-b4312d80-b8b3-11ea-97db-aaa52cb4cce5.png)

PC for comparison where the bug doesn't seem to be present.
![image](https://user-images.githubusercontent.com/10162026/85934666-391c4700-b8b4-11ea-82d1-845b1ec38a71.png)
![image](https://user-images.githubusercontent.com/10162026/85934672-4802f980-b8b4-11ea-8947-359a7e8b4e1c.png)

Also I suggest adding a check or at least a clarification that only those born after the **_happening_** magically turn 18 at the moment of conception, and that people born before that day (or at least those who were of legal age? Can't recall at the moment what the lore said, if anything, about what became of the rest of the world's children) remained the same and continued aging normally (except for the PC, since they were Gordon Freeman'd back into the world). It just looks a bit weird right now.

**Comment by serveral1** [comment]
> For Lilaya, this is intentional. As a half-demon, she can _appear_ as a different age than her actual age.

I'd be okay with this explanation if it wasn't for the fact that there's still an **_obvious_** mismatch between the birth dates and their actual age. I mean born in 94 and she's somehow 46 already? Bro I'm 26 and I was born in the same year, and by 2023 (which is the current save's date) I'd be 29, so clearly not even in my early 30s.

> The selfie description has also recently been updated and now only states your date of birth and age.

Yes, I've noticed that and I'm glad the issue is being worked on.

**Comment by AceXP** [comment]
Yes, but unlike you, Lilaya was born in the other world, so she was **_born as an 18 year old_** in '94. Add another 29 years and you get 46, since it's not her birthday yet.

As a side note, there is an option to switch off apparent age in the game options, so then both ages would be the same (46).

**Comment by serveral1** [comment]
I have no issue with the apparent age setting for the game, since it makes sense with the whole demonic age changing thing. My actual issue is really with that tacked on **_born as an 18 year old_** thing feeling more like a cope for the spaghettification of the age code that got pushed some 5 months ago or so. Before that, the actual and apparent age displays worked just fine.

Also there was never "another world", and to not get into further spoiler talk, I'll just say that the proof is with Lyssieth's explanation. Hell, in actuality Lilaya wasn't even supposed to be born in 1994, older saves show her birth date being in 1980, but because that would cause more conflicts everything is done "procedurally" now and all NPCs gets slapped with a lore breaking (at least in it's current form regarding the lore) additional 18 years to their actual ages.

**Comment by AceXP** [comment]
Yes, well I know about the Lyssieth explanation and all that. And I kind of agree with you and I don't. I think that your reasoning would be that Lily was born in our world, so she should be born earlier. But there is no Lily in the world as it is now, only Lilaya. That leaves the MC as the one and only exception to the 'born as an 18 year old' rule. As all other characters only exist in the world as it is now, the rule got applied retroactively to all of them. And yes, that is confusing, but I don't think it will be changed anytime soon.

**Comment by serveral1** [comment]
>But there is no Lily in the world as it is now, only Lilaya.
>As all other characters only exist in the world as it is now, the rule got applied retroactively to all of them.

It's not just Lily/Lilaya. Arthur, Rose and a bunch of other non-procedurally generated NPCs are also affected by this, and most of them aren't even Demons so the rule has an even more jarring effect on those characters since it doesn't carry any physicality with it. It's literally just text.

This is part Arthur's description for instance: "With messy brown hair, pale skin, and a thin frame, the Arthur of this world looks exactly the same as the one you've always known. Just as he was in your world...". Technically he's not a demon so why shouldn't his apparent age also be the same as his actual age? Technically speaking he's only aged the 3 of those years the MC was away, and not the other 18.

> That leaves the MC as the one and only exception to the 'born as an 18 year old' rule.

Well, prior to the whole age reformatting thing, everyone but those born after Lilith's manifestation were part of that exception, and the only difference in age between those NPCs and the MC where those 3 years where the MC was stuck in limbo, something which is quickly explained when you check the calendar. However the 18 years old thing is not, and it just ends up causing a lot of confusion, which is why I think this is still an issue worth addressing.

I think I'm somewhat aware as to why Inno decided to make these changes though, which in my guess are supposed to address the whole game start date thing (the year is always the same as the year your started your game. i.e and game started in 2018 would've ported the MC into 2021, etc. So say a game started in 2030 with the previous age code would've made Lilaya 50 years old, instead of 40ish which is what the lore says) and how it interacted with the previous constant values for some of the NPCs' birth dates, that would in turn end up causing a lot of lore/date conflicts in the long term. 

However there's a very simple fix for that: Making the start date a constant. Lot's of developers (specially single player devs) choose to do this since they all know the tech they use as a template for the world they're building will most likely not hold up with the passage of time (even for sci-fi settings), like who knows if in 5 years or so the MC's smartphone will be looked the same as how people currently look at flip-phones in older media. So imho, Inno should just pick a set date for when the game should start, instead of having the game grab whatever current year it is on your PC. Should save them from a couple of headaches in the future.

**Comment by serveral1** [comment]
Also, excuse the walls of text buy I guess plot-holes and roleplay inconsistencies are kind of my trigger... 😞

**Comment by AceXP** [comment]
I understand what you are saying, but it is up to @Innoxia if anything should be changed for this.

In my opinion, it isn't a bug, it is just the way things work in the world.
